### DISCUSSION TITLE
Tearing in Vulkan apps.

### DISCUSSION THREAD

**Discussion by somewhatfrog** [i3/i3#5285] [Q&A]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Vulkan apps with Vsync ON have tearing in windowed, tiled and i3's fullscreen mode, but don't have it if fullscreen is set in app itself. If app is set to fullscreen and then forced out of it with i3's shortcut for fullscreen mode, tearing appears.
## Expected Behavior

No tearing with Vsync
## Reproduction Instructions

1. Start any vulkan/dxvk/vkd3d app
2. Turn on Vsync in it or force with libstrangle or mangohud
3. Wait for tearing to appear (for me it is always at top 1/7th of the app window/tile/screen)
## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 
Binary i3 version:  4.21.1 (2022-10-24) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.21.1 (2022-10-24) (pid 626)
Loaded i3 config:
  /home/hello/.config/i3/config (main)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<details><summary>Happens with default config as well.</summary><pre>
</pre>
</details>

<pre>
Logfile URL:
</pre>

<pre>
- Linux Distribution & Version: arch 6.0.8-zen1-1-zen
- Are you using a compositor (e.g., xcompmgr or compton): I have picom, but I quit it when I use Vulkan apps.
- Nvidia proprietary, happens on all accessible versions as well as nvidia-open.
- 60hz screen, Vulkan apps (games) are locked to 60fps, proton and native.
- Happens regardless of early loading.
</pre>

Sorry in advance if this is not related to i3 or it is wrong place to report about this issue.

**Comment by stapelberg** [comment]
I don’t see how this could be related to i3. 

If the application needs to know about fullscreen being enabled to enable an optimized rendering path (just speculating), it should be able to use the standard way to listen for fullscreen notifications (via X11 window properties, see `NET_WM_STATE_FULLSCREEN`).

In general, i3 is not involved in rendering, drawing, or anything in that area. i3 only re-positions X11 windows. 

I’m not sure where you would need to report this issue, but it’s most likely not here.
